### PR TITLE
GH1369: Implemented MSBuild for VS2017

### PR DESCRIPTION
**Pull request by phillipsj**
Created enums and logic for handling MSBuild 15/VS2017 for the MSBuild runner. Also created
unit test.

### CODE REVIEW

**Comment by rasmus** [comment]
Any progress on this?

**Comment by phillipsj** [comment]
I have a PR submitted and an Addin on MyGet you can temporarily use.  I
will post it up when I am at a computer.

**Comment by phillipsj** [comment]
@rasmus Here is it: https://www.myget.org/F/beta-cake-addins/api/v2

    PM> Install-Package Cake.MsBuild15 -Version 1.0.

**Comment by rasmus** [comment]
@phillipsj Thanks

**Comment by devlead** [comment]
relates to #1369

**Comment by devlead** [comment]
@phillipsj your changes have now been merged! Thanks for your contribution 👍

File: src/Cake.Common.Tests/Fixtures/Tools/MSBuildRunnerFixture.cs [hunk 1]
```diff
@@ -30,6 +30,8 @@ public MSBuildRunnerFixture(bool is64BitOperativeSystem)
             KnownMSBuildPaths.Add("/Program86/MSBuild/12.0/Bin/amd64/MSBuild.exe");
             KnownMSBuildPaths.Add("/Program86/MSBuild/14.0/Bin/MSBuild.exe");
             KnownMSBuildPaths.Add("/Program86/MSBuild/14.0/Bin/amd64/MSBuild.exe");
+            KnownMSBuildPaths.Add("/Program86/Microsoft Visual Studio/2017/Enterprise/MSBuild/15.0/Bin/MSBuild.exe");
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Fixtures/Tools/MSBuildRunnerFixture.cs]: I think that we should add some tests for community and professional as well here.

File: src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs [hunk 2]
```diff
@@ -115,6 +115,23 @@ public void Should_Get_Correct_Path_To_MSBuild_Version_35(MSBuildToolVersion ver
             }
 
             [Theory]
+            [InlineData(MSBuildToolVersion.VS2017, PlatformTarget.x64, false, "/Program86/Microsoft Visual Studio/2017/Enterprise/MSBuild/15.0/Bin/amd64/MSBuild.exe")]
+            [InlineData(MSBuildToolVersion.VS2017, PlatformTarget.x86, false, "/Program86/Microsoft Visual Studio/2017/Enterprise/MSBuild/15.0/Bin/MSBuild.exe")]
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs]: I think that we should add some tests for community and professional as well here.

**Comment by patriksvensson** [comment]
Ooops, didn't realize this already was merged 😄

**Comment by phillipsj** [comment]
Ok will do.

**Comment by phillipsj** [comment]
I can submit another PR ;)

**Pull request merged**